common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java

```diff
@@ -554,4 +564,58 @@ public GetTagsResponse querySchemaTags(String schemaDefinition, String schemaNam
         }
         return getTagsResponse;
     }
+
+    /**
+     * AWS SDK Request interceptor that adds additional data to the UserAgent of Glue API requests.
+     */
+    @VisibleForTesting
+    protected class UserAgentRequestInterceptor implements ExecutionInterceptor {
+        private static final String ONE = "1";
+        private static final String ZERO = "0";
+
+        @Override
+        public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttributes executionAttributes) {
+            if (!(context.request() instanceof GlueRequest)) {
+                //Only applies to Glue requests.
+                return context.request();
+            }
+
+            GlueRequest request = (GlueRequest) context.request();
+            AwsRequestOverrideConfiguration overrideConfiguration =
+                request.overrideConfiguration().map(config ->
+                    config
+                        .toBuilder()
+                        .addApiName(getApiName())
+                        .build())
+                    .orElse((AwsRequestOverrideConfiguration.builder()
+                        .addApiName(getApiName())
+                        .build()));
+
+            return request.toBuilder().overrideConfiguration(overrideConfiguration).build();
+        }
+
+        private ApiName getApiName() {
+            return ApiName.builder()
+                .version(com.amazonaws.services.schemaregistry.common.MavenPackaging.VERSION)
+                .name(buildUserAgentSuffix())
+                .build();
+        }
+
+        private String buildUserAgentSuffix() {
+            Map<String, String> userAgentSuffixItems = ImmutableMap.of(
+                "autoreg", glueSchemaRegistryConfiguration.isSchemaAutoRegistrationEnabled() ? ONE : ZERO,
```

Comment:
nit : any specific reason or limit that they have to be short and small case ?

Reply:
Kept it short to not consume too many bits. No other specific reason. Lower case to be consistent with other components of the `UserAgent` 
```
vendor/Homebrew io/sync http/UrlConnection cfg/retry-mode/legacy
```